src/NHibernate.Test/CacheTest/QueryCacheFixture.cs

```diff
@@ -60,5 +60,19 @@ public void QueryCacheWithNullParameters()
 				s.Flush();
 			}
 		}
+
+		[Test]
+		public void QueryCacheWithScalarFunctionThatReturnsInt()
+		{
+			using (ISession s = OpenSession())
+			{
+				int result = s
+					.CreateSQLQuery("SELECT 200012")
```

Comment:
This query appears to be not compatible with all tested databases.

I think a fix would be:

 - Add an `override OnSetUp()` and move into it the following code from the other test:
```c#
using (ISession s = OpenSession())
{
	s.Save(simple, 1L);
	s.Flush();
}
```
- Add an `override OnTearDown()` and move into it the following code from the other test:
```c#
using (ISession s = OpenSession())
{
	s.Delete("from Simple");
	s.Flush();
}
```
- Change the SQL query to
```SQL
select 200012 from Simple where id_ = 1
```

I am unsure all databases will support the raw selected scalar, maybe you would need to change it to
```SQL
select 200011 + id_ as raw_scalar from Simple where id_ = 1
```
But that will turn it into a `long`, which would then require to adapt the test and check the bug is also demonstrated with that.

Otherwise you may restrict your test to run only on some databases, with something like:
```c#
if (!(Dialect is MsSql2000Dialect))
	Assert.Ignore(
		"Test use a raw SQL query that may not be compatible with currently tested database.");
```
But we favor adjusting tests for them to be run-able on all databases.